scripts/gen-netatt.sh

```diff
@@ -15,30 +15,52 @@
 # under the License.
 set -ex
 
-if [ -z "${DEPLOY_DIR}" ]; then
-    echo "Please set DEPLOY_DIR"; exit 1
-fi
+function check_var_setted () {
+  if [[ ! -v $1 ]]; then
+    echo "Please set $1"; exit 1
+  fi
+}
+
+check_var_setted DEPLOY_DIR
 
 if [ ! -d ${DEPLOY_DIR} ]; then
     mkdir -p ${DEPLOY_DIR}
 fi
 
-if [ -z "${INTERFACE}" ]; then
-    echo "Please set INTERFACE"; exit 1
-fi
-
-if [ -z "${VLAN_START}" ]; then
-    echo "Please set VLAN_START"; exit 1
-fi
-
-if [ -z "${VLAN_STEP}" ]; then
-    echo "Please set VLAN_STEP"; exit 1
-fi
+check_var_setted INTERFACE
+check_var_setted INTERFACE_DATA
+check_var_setted INTERFACE_MANAGEMENT
+check_var_setted INTERFACE_EXTERNAL
+check_var_setted INTERNALAPI_VLAN
+check_var_setted STORAGE_VLAN
+check_var_setted TENANT_VLAN
+check_var_setted INTERNALAPI_NET
+check_var_setted STORAGE_NET
+check_var_setted TENANT_NET
+check_var_setted INTERNALAPI_IP_START
+check_var_setted INTERNALAPI_IP_END
+check_var_setted STORAGE_IP_START
+check_var_setted STORAGE_IP_END
+check_var_setted TENANT_IP_START
+check_var_setted TENANT_IP_END
 
 echo DEPLOY_DIR ${DEPLOY_DIR}
 echo INTERFACE ${INTERFACE}
-echo VLAN_START ${VLAN_START}
-echo VLAN_STEP ${VLAN_STEP}
+echo INTERFACE_DATA ${INTERFACE_DATA}
+echo INTERFACE_MANAGEMENT ${INTERFACE_MANAGEMENT}
+echo INTERFACE_EXTERNAL ${INTERFACE_EXTERNAL}
+echo INTERNALAPI_VLAN ${INTERNALAPI_VLAN}
+echo STORAGE_VLAN ${STORAGE_VLAN}
+echo TENANT_VLAN ${TENANT_VLAN}
+echo INTERNALAPI_NET ${INTERNALAPI_NET}
+echo STORAGE_NET ${STORAGE_NET}
+echo TENANT_NET ${TENANT_NET}
+echo INTERNALAPI_IP_START ${INTERNALAPI_IP_START}
+echo INTERNALAPI_IP_END ${INTERNALAPI_IP_END}
+echo STORAGE_IP_START ${STORAGE_IP_START}
+echo STORAGE_IP_END ${STORAGE_IP_END}
+echo TENANT_IP_START ${TENANT_IP_START}
+echo TENANT_IP_END ${TENANT_IP_END}
 
 cat > ${DEPLOY_DIR}/ctlplane.yaml <<EOF_CAT
 apiVersion: k8s.cni.cncf.io/v1
```

Comment:
I think there are different things.
In this case during the gen-netatt.sh the NetworkAttachmentDefinition resources are being created. This will determine the IPs that will have the pods that have an interface on any of those networks. So at the end this script will generate openshift resources.

The newtork-attachment-definition.sh is useful to modify the crc VM (but gen-netatt.sh will be used regardless if we use crc or dev-metal3 deployment)